### PR TITLE
Adding Tetrahedral Bilinear Integrator Tests [tet-bl-tests-dev]

### PR DESCRIPTION
**Pull request by mlstowell**
Part of the reasoning behind this is just to increase test coverage.  The other reason is to verify correct functioning of various operators on tetrahedral elements in preparation for PR #1046 .

 | PR | Author | Editor | Reviewers | Assignment | Approval | Merge| 
 | --- | --- | --- | --- | --- | --- | --- | 
| [#1286](https://github.com/mfem/mfem/pull/1286) | @mlstowell | @tzanio | @delyank + @jamiebramwell | 02/22/20 | 03/12/20 | 03/22/20 | |

### CODE REVIEW

**Comment by mlstowell** [comment]
The high number of lines changed is due to the loops over element type which lead to greater indentation in most lines.  The actual number of meaningful lines changes is closer to 200.

File: fem/bilininteg.hpp [hunk 1]
```diff
@@ -643,10 +643,12 @@ class MixedVectorDivergenceIntegrator : public MixedScalarVectorIntegrator
              "vector field";
    }
 
+   // Subtract one due to the divergence and add one for the coefficient
+   // which is assumed to be at least linear.
    inline virtual int GetIntegrationOrder(const FiniteElement & trial_fe,
                                           const FiniteElement & test_fe,
                                           ElementTransformation &Trans)
-   { return trial_fe.GetOrder() + test_fe.GetOrder() + Trans.OrderW() - 1; }
+   { return trial_fe.GetOrder() + test_fe.GetOrder() + Trans.OrderW() - 1 + 1; }
```
Comment (delyank) [inline fem/bilininteg.hpp]: Why this way instead of deleting the `-1`?

File: fem/bilininteg.hpp [hunk 2]
```diff
@@ -840,6 +847,13 @@ class MixedWeakGradDotIntegrator : public MixedScalarVectorIntegrator
              "and the test space must be a vector field with a divergence";
    }
 
+   // Subtract one due to the gradient and add one for the coefficient
+   // which is assumed to be at least linear.
+   inline virtual int GetIntegrationOrder(const FiniteElement & trial_fe,
+                                          const FiniteElement & test_fe,
+                                          ElementTransformation &Trans)
+   { return trial_fe.GetOrder() + test_fe.GetOrder() + Trans.OrderW() - 1 + 1; }
+
```
Comment (delyank) [inline fem/bilininteg.hpp]: Same as before

File: fem/bilininteg.hpp [hunk 3]
```diff
@@ -680,6 +682,11 @@ class MixedScalarWeakGradientIntegrator : public MixedScalarIntegrator
              "and the test space must be H(Div)";
    }
 
+   inline virtual int GetIntegrationOrder(const FiniteElement & trial_fe,
+                                          const FiniteElement & test_fe,
+                                          ElementTransformation &Trans)
+   { return trial_fe.GetOrder() + test_fe.GetOrder() + Trans.OrderW() - 1; }
+
```
Comment (delyank) [inline fem/bilininteg.hpp]: Why no `+1` for the coefficient here?

**Comment by delyank** [comment]
The modification, as far as the library itself is concerned, is very minor and likeable. It makes more sense this way and improves the code. I don't see a reason not to merge it and avoid delaying the other PR.

**Comment by tzanio** [comment]
@jamiebramwell, do you need more time to review this?

**Comment by jamiebramwell** [comment]
I apologize for the delay! I'll take a look at it now.

File: fem/lininteg.hpp [hunk 4]
```diff
@@ -281,10 +281,13 @@ class VectorFEBoundaryFluxLFIntegrator : public LinearFormIntegrator
 private:
    Coefficient *F;
    Vector shape;
+   int oa, ob;
```
Comment (jamiebramwell) [inline fem/lininteg.hpp]: Some documentation of these parameters would be appreciated.

**Comment by tzanio** [comment]
@mlstowell, can you address @delyank's and @jamiebramwell's (small) comments?

**Comment by tzanio** [comment]
Can you please resolve the merge conflicts?

**Comment by tzanio** [comment]
Merged in `next` for testing...

**Comment by tzanio** [comment]
LGTM

**Pull request merged**